dev/examples.el

```diff
@@ -692,7 +692,9 @@ new list."
 
   (defexamples -distinct
     (-distinct '()) => '()
-    (-distinct '(1 2 2 4)) => '(1 2 4)))
+    (-distinct '(1 2 2 4)) => '(1 2 4)
+    (-distinct '(t t t) => '(t))
+    (-distinct '(nil nil nil) => '(nil))))
```

Comment:
We should also add 3 more versions for these with `(let ((-compare-fn #'eq)) BODY-HERE)` setup so we actually hit the new code with the hash tables.

Reply:
Done.